Simplenote/src/main/java/com/automattic/simplenote/utils/TagsMultiAutoCompleteTextView.java

```diff
@@ -83,7 +82,7 @@ public void init() {
     }
 
     public void notifyTagsChanged() {
-        String lexical = getText().toString().trim();
+        String lexical = getText().toString().replace(',', ' ').trim();
```

Comment:
this makes me nervous, since we're making a change in a seemingly random place without any explanation as to why we're swapping spaces in for the commas.

what's the purpose here anyway? why are we doing that? is it to make sure that entered tags are split by the comma? if so, maybe we leave in the commas and handle them wherever they get split?

won't the spaces here mess up the canonical tag name anyway? `getCanonicalFromLexical` expects a valid tag name, of which one with spaces in it is not.

Reply:
This is how basically tags are being added in AppCompatMultiAutoCompleteTextView. AppCompatMultiAutoCompleteTextView is the TextView at the bottom of the Note Editor view, which is used to add tags to a note. I've inlined the method calls for brevity. The following is the original code without any of my changes.

       public void onTextChanged(CharSequence s, int start, int before, int count) {
            if (count >= 1 && s.charAt(start) == SPACE) {
                  if (TagUtils.hashTagValid(s.toString())) {
                      String lexical = getText().toString().trim();
                      String canonical = TagUtils.getCanonicalFromLexical(mBucketTag, lexical);
                      notifyTagsChanged(canonical);
            }
       }

getText() returns the text that TagsMultiAutoCompleteTextView  is displaying. Based on the condition, `s.charAt(start) == SPACE`, it would always return a text that has a leading whitespace. 

That's why trim() is being called later, so to remove that leading whitespace, like:

    String lexical = getText().toString().trim();

I only changed the following two lines of code.
```
if (count >= 1 && s.charAt(start) == SPACE) 
// TO
if (count >= 1 && (s.charAt(start) == ' ' || s.charAt(start) == ','))
```

With the refactored condition, the returned text would never contain any commas in the middle, in addition to whitespaces. So when a user enters either whitespace or a comma, it would try to save the text as a tag.

```
String lexical = getText().toString().trim();
// TO
String lexical = getText().toString().replace(',', ' ').trim();
```
trim() here trims the leading whitespace when 'space' is used as the delimiter.
And the purpose of `replace(',', ' ') is to trim the leading comma when a comma is being used as the delimiter.

The other changes made to AddTagActivity, TagDialogFragment, strings.xml are for the 'Add Tag' view.

Reply:
@dmsnell Please do let me know if it's still not clear 🙂